include/RAJA/policy/atomic_builtin.hpp

```diff
@@ -40,8 +40,47 @@ struct builtin_atomic {
 namespace detail
 {
 
+template <std::size_t BYTES>
+struct BuiltinAtomicTypeImpl {
+  static_assert(!(BYTES == sizeof(unsigned) ||
+                  BYTES == sizeof(unsigned long long)),
+                "Builtin atomic operations require targets that match the size of 'unsigned int' or 'unsigned long long' (usually 4 or 8 bytes).");
+};
+
+template <>
+struct BuiltinAtomicTypeImpl<sizeof(unsigned)> {
+  using type = unsigned;
+};
+
+template <>
+struct BuiltinAtomicTypeImpl<sizeof(unsigned long long)> {
+  using type = unsigned long long;
+};
+
+template <class T>
+using BuiltinAtomicType = typename BuiltinAtomicTypeImpl<sizeof(T)>::type;
+
 #if defined(RAJA_COMPILER_MSVC) || (defined(_WIN32) && defined(__INTEL_COMPILER))
 
+RAJA_DEVICE_HIP
```

Comment:
Do we need the `RAJA_DEVICE_HIP` macro on these methods given the preceeding `#if defined...` guard? What am I missing?

Reply:
I was just following the pattern of the existing methods inside that if guard. I'm happy to remove RAJA_DEVICE_HIP on all of them, but I have no idea what that macro actually does.

Reply:
It's defined in the file at line 26. @rchen20 added it. We should check with him.

Reply:
Apparently, advanced versions of NVCC complained about host_device decorations on these functions. https://github.com/LLNL/RAJA/pull/721 This change removed the decorations for newer CUDA's, and required re-adding the decorations for HIP. Because this macro is only used in atomic_builtin.hpp, I think it's pretty safe to keep around.

Reply:
I understand why it would be needed for CUDA or HIP, but these decorations are used in a section for MSVC or Intel Windows build. I'm asking why they are needed for those use cases.

Reply:
You're right, we could remove the use of this macro from this particular MSVC && Intel block.

Reply:
I removed it from all the functions in this block.